frontends/mit-open/src/page-components/ResourceCarousel/types.ts

```diff
@@ -29,6 +29,7 @@ type TabConfig<D extends DataSource = DataSource> = {
   label: React.ReactNode
   cardProps?: CardProps
   data: D
+  eager?: boolean
```

Comment:
is this used?

Reply:
Looks like not currently used.

On production right now, the carousels only load the data once the tab is clicked into. With this PR, they will load all the tab data immediately (eagerly). So we'll get a bunch more API requests—for example, the homepage has 7 carousel tabs, so it will make 7 API requests on load. Similarly, the dashboard page has lots of carousels. James and I discussed the idea of making the eager behavior opt-in.

@jkachel You probably have a better sense than I do whether all those parallel API requests would be problematic (for the server). I'm happy to do always-eager and make opt-in later if seems appropriate.

Reply:
This is set up so only the 4 in the Featured tab eager load - the rest lazy load. So, you _should_ only get 4 from the featured carousel plus single ones for the other carousel(s), and the featured ones shouldn't have to reload if you select tabs (since React Query should cache that response if it's successful). 

FWIW, and the code sorta gets halfway there now, we can use a similar pattern to the DataPanel to eager load the tab buttons; if we do that, we can add the eager toggle there. (It seemed more like Featured was special in this regard so didn't go all the way with it here.) 

(I was working on making this flaggable with that prop but I went a different route instead. That's a bit I missed from that so did in https://github.com/mitodl/mit-open/pull/1169/commits/edef882e488616db29fb9606afb0debf3933a994 .)